### DISCUSSION TITLE
Multiple checkboxes bound to a model cannot be unchecked?

### DISCUSSION THREAD

**Discussion by ryanscherler** [livewire/livewire#3069] [7. Bugs (archived)] [answered]
The following code seems to not allow a user to 'uncheck' a checkbox if its been filled by a model (or any data really). It seems that as soon as a user tries to uncheck the input, it immediately is checked again if it exists in the model. Any ideas how to prevent this?

```php
... Component {
    public $roles = [];

    public function mount(User $user)
    {
        $this->roles = $user->roles->pluck('id')->all();
    }
...
}

<div>
    <input type="checkbox" wire:model="roles" value="1">
    <input type="checkbox" wire:model="roles" value="2">
    <div>@json($roles)</div>
</div>
```

![checkboxes-with-model](https://user-images.githubusercontent.com/881938/77969883-ad3c5400-729f-11ea-9884-9005844077fd.gif)

Reference to original closed issue: https://github.com/livewire/livewire/issues/173#issuecomment-527915573

**Comment by JayBizzle** [comment]
I have just come across a similar issue, but mine was slightly different to yours (see screen recording).

![livewire](https://user-images.githubusercontent.com/340752/78183198-64cb9480-745f-11ea-92e4-5bc9bc8c54fe.gif)


```php
    public $checkbox_values = [];

    public function mount()
    {
        $this->checkbox_values = [3130, 3129];
    }
```

If I change the `checkbox_values` array to an array of string values, it then works as expected.

```php
    public $checkbox_values = [];

    public function mount()
    {
        $this->checkbox_values = ['3130', '3129'];
    }
```

The checkbox code...
```blade
<input type="checkbox" wire:model="checkbox_values" value="{{ $blog->id }}">
```

@ryanscherler I notice your `@json` output contains an `int` and a `string`. I think our issues are related. Thoughts?

**Comment by ryanscherler** [comment]
@JayBizzle Interesting! This does seem related. I imagine that when I fill the values from the model, the id is an `int` then when applied via the checkbox its a string. Can I (or should I have to) somehow cast the ID from the model into a string when filling to ensure they are always strings?

**Comment by JayBizzle** [comment]
You could do something like this...

```php
$this->roles = array_map('strval', $user->roles->pluck('id')->all());
```

Give that a try and see if your issue is resolved.

Can't decide if this is a bug, or something we have to live will.

Guess we will have to wait for @calebporzio to give his thoughts.

👍

**Comment by ryanscherler** [comment]
That totally worked. No idea why an integer value would mess things up - but its all good now!

**Comment by lancepioch** [comment]
Glad you got it figured out :)

**Comment by JayBizzle** [comment]
@lancepioch So, are we saying this isn't a bug?!

**Comment by roni-estein** [comment]
> @lancepioch So, are we saying this isn't a bug?!

I would think that this doesn’t fit with ease of use paradigm. If it’s possible it should be left as a bug with a hopeful resolution or a clear notation in the docs. Otherwise this will add precisely 19 grey hairs to each dev who runs into it.

**Comment by ryanscherler** [comment]
I totally agree - without the suggestion from @JayBizzle I would have had no idea to solve it that way. At the least maybe a note in the docs regarding integers being an issue with pairing to model values?

**Comment by jartaud** [comment]
> You could do something like this...
> 
> ```
> $this->roles = array_map('strval', $user->roles->pluck('id')->all());
> ```
> 
> Give that a try and see if your issue is resolved.
> 
> Can't decide if this is a bug, or something we have to live will.
> 
> Guess we will have to wait for @calebporzio to give his thoughts.

Bug still exists. Thanks @JayBizzle

**Comment by GertjanRoke** [comment]
@JayBizzle which version of Livewire did you use to test this? and maybe share the code you test it with?

**Comment by j2L4e** [comment]
https://github.com/alpinejs/alpine/commit/3939fb722d31a65ebf43fe1e8dc2448344cedaf9 the `.number` modifier has been part of alpine quite a while (since 2.4.0), shouldn't it "just work" with livewire then?

**Comment by vguerrerobosch** [comment]
> > @lancepioch So, are we saying this isn't a bug?!
> 
> I would think that this doesn’t fit with ease of use paradigm. If it’s possible it should be left as a bug with a hopeful resolution or a clear notation in the docs. Otherwise this will add precisely 19 grey hairs to each dev who runs into it.

Here you go! Add 19 more to the total.

**Comment by Tylerian** [comment]
This is still a known bug and I've just faced it today.

Any chances it's gonna be fixed anytime soon?

**Comment by zepfietje** [comment, marked as answer]
This bug will be fixed by https://github.com/livewire/livewire/pull/4382.

**Reply by zepfietje** [reply]
@joshhanley this discussion can be marked as answered!

**Reply by joshhanley** [reply]
@zepfietje thanks! Yeah will do once it has been tagged and released, which won't be until after the break now 🙂
